src/templates/console.html

```diff
@@ -19,39 +19,64 @@
           pushCode,
           {
             greetings: "Welcome to the Pyodide terminal emulator 🐍",
-            prompt: "[[;red;]>>> ]"
+            prompt: "[[;blue;]>>> ]"
           }
         );
 
+        function trim(string) { return string.replace(/\n$/, "") }
+
+        term.stdout = { write: function (string) { term.echo(trim(string)) } }
+        term.stderr = { write: function (string) { term.error(trim(string)) } }
+
         window.term = term;
         pyodide.runPython(`
           import io, code, sys
-          from js import term, pyodide
+          from js import term
+
+
+          class CallbackBuffer(object):
+              """
+              Internal StdStream buffer that trigger flush callback.
+              """
+              def __init__(self, flush_callback):
+                  self._flush_callback = flush_callback
+                  self.closed = False
+              def writable(self): return True
+              def seekable(self): return False
+              def write(self, data):
+                  self._flush_callback(data.tobytes().decode())
+                  return len(data)
+
+
+          class StdStream(io.TextIOWrapper):
+              """
+              Custom std stream to retdirect sys.stdou/stderr to terminal.
+              """
+              def __init__(self, flush_callback):
+                  buffer = io.BufferedWriter(CallbackBuffer(flush_callback))
+                  super().__init__(buffer, line_buffering=True)
```

Comment:
Could you please move these 3 classes to `src/pyodide.py`,  keep them private (e.g. `_ConsoleStdStream`, '_ConsoleCallbackBuffer', `_Console`) and add some unit tests for them in `src/tests/test_pyodide.py`. Eventually we might make an actual pyodide Python package with multiple modules, but for now I thing putting everything in a single file is fine.

Reply:
I agree that this should be tested but do you think `src/pyodide.py` is the better place to put REPL stuff?

Reply:
Otherwise we would need to address https://github.com/iodide-project/pyodide/issues/896 first and then put it under `pyodide/console.py`. Maybe that would be better indeed.

Reply:
Answered to #896 where your proposal makes sense. However, should this be blocking for this PR?